frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,210 @@
+          resources: {},
+          volumeMounts: [],
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      name: podToDebug?.metadata?.name,
+      tolerations: podToDebug?.spec?.tolerations || [],
+      nodeSelector: podToDebug?.spec?.nodeSelector || {},
+    },
+  };
+};
+
+const DebugTerminalError: React.FC<DebugTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const DebugTerminalInner: React.FC<DebugTerminalInnerProps> = ({ pod, activeContainer }) => {
+  const { t } = useTranslation();
+  const infoMessage = (
+    <div className="co-terminal-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  const [podData] = pod;
+  switch (podData?.status?.phase) {
+    case 'Failed':
+      return (
+        <DebugTerminalError
+          error={
+            <>
+              {t('public~The debug pod failed. ')}
+              {podData?.status?.containerStatuses?.[0]?.state?.terminated?.message ||
+                podData?.status?.message}
+            </>
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader obj={podData} activeContainer={activeContainer} infoMessage={infoMessage} />
+      );
+    default:
+      return <LoadingBox />;
+  }
+};
+
+export const DebugTerminal: React.FC<DebugTerminalProps> = ({ pod, containerName }) => {
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [podData] = pod;
+  const podName = podData?.metadata.name;
+  const podNamespace = podData?.metadata.namespace;
+  const podContainerName = containerName || podData?.spec.containers[0].name;
+  const debugPodName = `${podName}-${podContainerName}-debug`;
+  React.useEffect(() => {
+    const deleteDebugPod = async (podToDelete) => {
+      try {
+        await k8sKillByName(PodModel, podToDelete, podNamespace);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Could not delete container terminal debug pod.', e);
+      }
+    };
+    const createDebugPod = async () => {
+      try {
+        const podToCreate = getDebugPod(debugPodName, podData, podContainerName);
+        await k8sCreate(PodModel, podToCreate);
+      } catch (e) {
+        setErrorMessage(e.message);
+      }
+    };
+    createDebugPod();
+    window.addEventListener('beforeunload', deleteDebugPod);
+    return () => {
+      deleteDebugPod(debugPodName);
+      window.removeEventListener('beforeunload', deleteDebugPod);
+    };
+  }, [debugPodName, podContainerName, podData, podName, podNamespace]);
```

Comment:
I think this will still create a new pod when `podData` changes, right?

We really want to only create the pod once. If we want to switch containers, we should create the debug pod with all containers and just connect to the right one.

Reply:
Investigating switching the active container in the debug pod.

Reply:
So, looking at the 3.11 code, we only had the one container that we were debugging.  Instead of switching dynamically, I think I will just simplify to only having one container per debug pod.  Thoughts?

Reply:
Cleaned this up to ensure we are not creating etra pods.